docs/DeveloperGuide.md

```diff
@@ -1,34 +1,179 @@
+
+1. When a user enters a calendar-related command, the command is analysed by `CalendarCommandInterpreter`.
+1. Once determined, the relevant information (eg. semester, academic year) are extracted by `CalendarParser`.
+1. Then, either AddFirstSemester or AddSecondSemester class that corresponds the semester number is created. 
+1. Subsequently, it separates events by the required month and year in `CalendarList`
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
+Note that:
+* `acadamic year` is parsed into corresponding to only one year according to the semester in `EventParser` class.
+* Calendar view of the whole year is not available. Only semester 1 or 2 of an academic year can be viewed at a time.
+* Event name size must be less than 10 characters to be displayed neatly (current implementation), however
+it can be implement to truncate longer names to fit nicely
 
 ## Product Scope
 ### Target user profile
 
-{Describe the target user profile}
+Note that:
+* `attendance add` command requires a line-by-line insertion of the student attendance data. 
+The user is given an option to either use an existing list stored under StudentListCollection or
+create a new attendance list. `n/` and `p/` flags are used to insert new attendance.
 
-### Value proposition
+### Performance
+![Performance](images/Performance.png "Class diagram of Performance component")     
+*Class diagram of the Performance component*  
+1. When a user enters a performance-related command, the command is analysed by `PerformanceCommandInterpreter`. 
+1. Once determined, the relevant class that corresponds to the command is created (e.g. AddPerformance, 
+DeletePerformance...), and ask for relevant information (e.g. event name, student name, student result) from the user. 
+1. Then, with the information extracted from the previous step passed into it. It modifies PerformanceList` under
+the event class correspond to the input event name.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
+Note that:
+* All PerformanceList class should be created under an Event class. A PerformanceList class cannot exist 
+by its own. 
+* All Performance commands are line-by-line commands. This aims to assist the user with correct command format and
+prevent time wasted on key in wrong commands. 
+
+### Student List Collection
+![Student](images/Student.png "Class diagram of Student component")     
+*Class diagram of the Student component*  
+1. When a user enters an studentList-related command, the command is analysed by `StudentCommandInterpreter`. 
+1. Once determined, the relevant class that corresponds to the type of command is created.
+1. Then, the class will execute base on its function. It modifies `AttendanceList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
 
-{Describe the value proposition: what problem does it solve?}
+Note that:
+* studentList-related commands can be executed without the existence of events.
```

Comment:
I am not sure if you format this on purpose, but it seems that those parts look similar to `Event`, `Attendance`, and `Calendar Implementation`. Maybe these two parts also belong to `Feature Design and Implementation` section?